cosim/include/vcs/bsg_zynq_pl.h

```diff
@@ -99,23 +99,23 @@ class bsg_zynq_pl {
     } else if (address >= GP1_ADDR_BASE &&
                address <= GP1_ADDR_BASE + GP1_ADDR_SIZE_BYTES) {
       index = 1;
-      address = address - GP1_ADDR_BASE;
+      offset = (unsigned int)(address - GP1_ADDR_BASE);
     } else {
       bsg_pr_err("  bsg_zynq_pl: unsupported AXIL port %d\n", index);
     }
 
     bsg_pr_dbg_pl("  bsg_zynq_pl: AXI writing [%x] -> port %d, [%x]<-%8.8x\n",
-                  address_orig, index, address, data);
+                  address, index, offset, data);
 
     if (index == 0) {
-      axi_gp0->axil_write_helper(address, data, wstrb, tick);
+      axi_gp0->axil_write_helper(offset, (int)data, wstrb, tick);
     } else if (index == 1) {
-      axi_gp1->axil_write_helper(address, data, wstrb, tick);
+      axi_gp1->axil_write_helper(offset, (int)data, wstrb, tick);
     }
   }
 
   int32_t axil_read(uintptr_t address) {
-    uintptr_t address_orig = address;
+    unsigned int offset;
```

Comment:
Is this fixing a bug? When changing the code to current version I had to solve a lot of 32/64 issues for the different pynq boards